src/index.js

```diff
@@ -58,26 +58,34 @@ class ActiveStorageProvider extends React.Component<Props> {
     }
   }
 
+  _getHeaders = (): CustomHeaders => {
+    // fastest performance for indifferent access: https://jsperf.com/downcase-and-merge-object-keys
+    const mappedHeaders = {}, { headers = {} } = this.props
+    try {
+      Object.keys(headers || {}).reduce((c, k) => (c[k.toLowerCase()] = headers[k], c), mappedHeaders)
+    } catch { /* ignore invalid headers */ }
```

Comment:
Objects can only have strings for keys, so this should never throw. If I'm not mistaken, we don't need the try catch, in which case we can make the assignment of `mappedHeaders` (how about `normalizedHeaders`?) explicit, just for readability

```js
const { headers } = this.props
const normalizedHeaders = Object.keys(headers).reduce(
  (acc, key) => (acc[key.toLowerCase()] = headers[key]),
  {}
)
```

Reply:
it's actually possible to have non-strings as object keys (`{[() => 'test']: 'test' }`) though I suppose it wouldn't pass propTypes validations...

I mostly wanted to ensure it's not possible to break the request with invalid keys, but maybe we do want an error there

Reply:
Non-string keys are coerced into strings under the hood. JS is weird.

```js
Object.keys({[() => 'test']: 'test' }) // => [ "() => 'test'" ]
```

Symbols are an exception, ofc, but they're not included in the output of Object.keys() anyway

Reply:
though looking at it now, I realize that the `try...catch` block should be moved into the reducer, not as a wrapper